### PR TITLE
Add MFEM_VERSION definition to _config.hpp and config.mk [mfem-version-dev]

### PR DESCRIPTION
**Pull request by v-dobrev**
In both places MFEM_VERSION is an integer, e.g. version "3.3" -> 30300.

### CODE REVIEW

**Comment by v-dobrev** [comment]
Between releases, we can set the version to something like `"3.3.99" <-> 30399`.

**Comment by tzanio** [comment]
One minor thing -- is it possible to add a function in make that extract the major, minor and patch numbers from the MFEM version. 

I'd like `make info` to print `3.3.0 (30300)` if possible... the missing initial `0` I think can confuse some folks.

File: makefile [hunk 1]
```diff
@@ -392,7 +394,7 @@ help:
 	@true
 
 status info:
-	$(info MFEM_VERSION         = $(MFEM_VERSION))
+	$(info MFEM_VERSION         = $(MFEM_VERSION) [v$(MFEM_VERSION_STRING)])
```
Comment (tzanio) [inline makefile]: Thanks!

**Comment by tzanio** [comment]
@mlstowell and @aaroncblack do you have any objections to merge this?

**Comment by aaroncblack** [comment]
I prefer using separate variables, like MFEM_VERSION_MAJOR, MFEM_VERSION_MINOR, and MFEM_VERSION_PATCH.  It's easier for an occasional contributor to quickly understand.

**Comment by v-dobrev** [comment]
OK, I'll add definitions for those macros using `MFEM_VERSION`, e.g.
```c++
#define MFEM_VERSION_MAJOR ((MFEM_VERSION)/10000)
```

**Pull request merged**